README.md

```diff
@@ -1,707 +1,1799 @@
+```
+
+</details>
+
+<details><summary><b>Parameters</b></summary>
+
+| Name   | Description                  |
+| :----- | :--------------------------- |
+| `func` | The function to be memoized. |
+
+</details>
+
+<details><summary><b>Returns</b></summary>
+
+A memoized function with a `.clearCache()` method attached.
+
+</details>
+
+<details><summary><b>Type parameters</b></summary>
+
+| Name   | Description                                |
+| :----- | :----------------------------------------- |
+| `Func` | The type of the function that is memoized. |
+
+</details>
+
+<details><summary><b><code>Examples</code></b></summary>
+
+Prior to `weakMapMemoize`, you had this problem:
+
+```ts
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)
+
+parametricSelector(state, 0) // Selector runs
+parametricSelector(state, 0)
+parametricSelector(state, 1) // Selector runs
+parametricSelector(state, 0) // Selector runs again!
+```
+
+Before you could solve this in a number of different ways:
+
+1. Set the `maxSize` with [`defaultMemoize`]:
+
+```ts
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id),
+  {
+    memoizeOptions: {
+      maxSize: 10
+    }
+  }
+)
+```
+
+But this required having to know the cache size ahead of time.
+
+2. Create unique selector instances using [`useMemo`].
+
+```tsx
+const parametricSelector = (id: number) =>
+  createSelector([(state: RootState) => state.todos], todos =>
+    todos.filter(todo => todo.id === id)
+  )
+
+const MyComponent: FC<Props> = ({ id }) => {
+  const selectTodosById = useMemo(() => parametricSelector(id), [id])
+
+  const todosById = useSelector(selectTodosById)
+
+  return (
+    <div>
+      {todosById.map(todo => (
+        <div key={todo.id}>{todo.title}</div>
+      ))}
+    </div>
+  )
+}
+```
+
+3. Using [`useCallback`].
+
+```tsx
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)
+
+const MyComponent: FC<Props> = ({ id }) => {
+  const selectTodosById = useCallback(parametricSelector, [])
+
+  const todosById = useSelector(state => selectTodosById(state, id))
+
+  return (
+    <div>
+      {todosById.map(todo => (
+        <div key={todo.id}> {todo.title}</div>
+      ))}
+    </div>
+  )
+}
+```
+
+4. Use [`re-reselect`]:
+
+```ts
+import { createCachedSelector } from 're-reselect'
+
+const parametricSelector = createCachedSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)((state: RootState, id: number) => id)
+```
+
+Starting in 5.0.0, you can eliminate this problem using `weakMapMemoize`.
```

Comment:
:hand: If we're going to have this kind of comparison, there ought to be more information here explaining _why/how_ `weakMapMemoize` solves this issue.

Reply:
I will the additional info. Just to be sure, we are specifically refering to the "cache size" problem right?

Reply:
Yep!